src/components/Timeline/updates/elements/PrettyEmail.tsx

```diff
@@ -0,0 +1,174 @@
+      setEmailData(data);
+    }
+
+    parseEmailStr();
+  }, [emailStr]);
+
+  if (emailData) {
+    return (
+      <Box>
+        <EmailHeader headers={emailData.headers} />
+        <Typography style={{ fontWeight: 'bold', marginBottom: 8 }}>
+          {emailData.headers.Subject}
+        </Typography>
+        <Collapse
+          collapsedSize={COLLAPSED_SIZE}
+          in={(didMeasure && !needsCollapse) || !collapsed}
+        >
+          <div ref={divCallback}>
+            <EmailBody body={emailData.body} />
+          </div>
+        </Collapse>
+        {needsCollapse && (
+          <Button
+            color="primary"
+            onClick={() => setCollapsed(!collapsed)}
+            startIcon={collapsed ? <ExpandMore /> : <ExpandLess />}
+            style={{ textTransform: 'uppercase' }}
+          >
+            <FormattedMessage
+              id={collapsed ? 'misc.email.expand' : 'misc.email.collapse'}
+            />
+          </Button>
+        )}
+      </Box>
+    );
+  } else {
+    return <CircularProgress />;
+  }
+};
+
+const useBodyStyles = makeStyles<Theme, { plain: boolean }>((theme) => ({
+  body: {
+    '& blockquote': {
+      borderColor: theme.palette.text.disabled,
+      borderLeftWidth: 4,
+      borderStyle: 'solid',
+      borderWidth: 0,
+      marginLeft: 2,
+      opacity: 0.8,
+      paddingLeft: 10,
+    },
+    fontFamily: 'sans-serif',
+    whiteSpace: ({ plain }) => (plain ? 'pre' : 'normal'),
+  },
+}));
+
+const EmailBody: React.FC<{
+  body: LetterparserNode['body'];
+  forcePlain?: boolean;
+}> = ({ body, forcePlain = false }) => {
+  const plain = forcePlain || typeof body == 'string';
+  const classes = useBodyStyles({ plain });
+
+  if (Array.isArray(body)) {
+    let bodyToRender = body.find(
+      (bodyCandidate) => bodyCandidate.contentType.type === 'text/html'
+    );
+
+    if (!bodyToRender) {
+      bodyToRender = body[0];
+    }
+
+    return <EmailBody body={bodyToRender.body} forcePlain />;
+  } else {
+    const content = typeof body == 'string' ? body : body.toString();
+
+    // TODO: Sanitize content before rendering
+    return (
+      <div
+        className={classes.body}
+        dangerouslySetInnerHTML={{ __html: content }}
+      />
+    );
+  }
+};
+
+const EmailHeader: React.FC<{ headers: LetterparserNode['headers'] }> = ({
+  headers,
+}) => {
+  const RELEVANT_HEADERS = ['from', 'to', 'cc'];
+  return (
+    <Grid container direction="column">
+      {RELEVANT_HEADERS.map((headerName) => {
+        const matchedHeader = Object.entries(headers).find(
+          ([key]) => key.toLowerCase() == headerName.toLowerCase()
+        );
+
+        if (matchedHeader && matchedHeader[1]) {
+          const values = matchedHeader[1].split(',');
+
+          return (
+            <Grid key={headerName} item style={{ marginBottom: 8 }}>
+              <Grid container direction="row" wrap="nowrap">
+                <Grid item style={{ paddingRight: 12, paddingTop: '0.2em' }}>
```

Comment:
I think the grid supports a spacing or gap prop which handles this stuff. I would use the built in grid api to achieve this.

Reply:
It depends I guess if you're trying to do padding to provide spacing between components.

Reply:
Would that work even in this case where I need the padding/margin to be different in different directions?

Reply:
Thanks for explaining that @lowlandjuju! I found a `spacing` property that seems to do what you're suggesting, and I was able to remove one level of `<Grid>` by having the same one be both `container` and `item` which I hadn't realized was possible.